addons/calendar/models/res_users.py

```diff
@@ -10,6 +10,22 @@
 class Users(models.Model):
     _inherit = 'res.users'
 
+    calendar_default_privacy = fields.Selection(
+        [('public', 'Public'),
+         ('private', 'Private'),
+         ('confidential', 'Only internal users')],
+        'Calendar Default Privacy', default='public', required=True, store=True,
+        help="Default privacy setting for whom the calendar events will be visible."
+    )
+
+    @property
+    def SELF_READABLE_FIELDS(self):
+        return super().SELF_READABLE_FIELDS + ['calendar_default_privacy']
+
+    @property
+    def SELF_WRITEABLE_FIELDS(self):
+        return super().SELF_WRITEABLE_FIELDS + ['calendar_default_privacy']
+
```

Comment:
I am not a fan of adding fields on res.users because we try to remove them :-/. 
The issue is that when you access these kind of field, it will lock the row and it can be problematic for this particular table. 
Accessing the res.user row from calendar.event (you can have a tons of events) can be really bad.

Another possibility would be to merge the microsoft.calendar.credentials and google.calendar.credentials table inside a calendar.user.setting model and use that table for all the fields of calendar related to the user that should not be stored on the user.

But it will require an upgrade script and the best would be to create a task in higher priority to do that and after reopen this task and use the setting table. what do you think ?